pythonAPIClient/client.py

```diff
@@ -115,33 +135,157 @@ def register(self, owner, permissionLevel='user'):
             Permission level of the owner of the client application.
             Default permission level is 'user'.
 
-        Raises
-        ------
 
         Returns
         -------
         str
             Client id of the newly registered client application.
         str
             Client secret of the newly registered client application.
+
         """
-        pass
+        url = Client.base_URL + 'apps/register'
+        header = self.header_builder()
+        data = {'owner': owner, 'permissionLevel': permissionLevel}
+        response = self._check_token(response=requests.post(url, headers=header, data=data),
+                                     flag='POST', url=url)
```

Comment:
data should also be in the argument in this _check_token function call, or else if a new token should be acquired in the _check_token function, an empty request body will be sent

Reply:
same for other methods where there should be request body/query params